source/slang/slang-check-conversion.cpp

```diff
@@ -201,11 +201,137 @@ namespace Slang
         return baseStructDeclRef;
     }
 
+    bool SemanticsVisitor::_hasExplicitConstructor(StructDecl* structDecl)
+    {
+        bool result = false;
+        for (auto ctor : getMembersOfType<ConstructorDecl>(getASTBuilder(), structDecl, MemberFilterStyle::All))
+        {
+            // constructor that is not synthesized must be user defined.
+            if (ctor.getDecl()->findModifier<SynthesizedModifier>() == nullptr)
+            {
+                result = true;
+                break;
+            }
+        }
+        return result;
+    }
+
+    // TODO: We might need to find a good way to get the synthesized constructor instead of traversing all of constructors.
+    ConstructorDecl* SemanticsVisitor::_getSynthesizedConstructor(StructDecl* structDecl)
+    {
+        for (auto ctor : getMembersOfType<ConstructorDecl>(getASTBuilder(), structDecl, MemberFilterStyle::All))
+        {
+            if (ctor.getDecl()->findModifier<SynthesizedModifier>() != nullptr)
+            {
+                return ctor.getDecl();
+            }
+        }
+
+        return nullptr;
+    }
+
+    Expr* SemanticsVisitor::_prepareCtorInvokeExpr(Type* toType, const SourceLoc& loc, const List<Expr*>& coercedArgs)
+    {
+        auto* varExpr = getASTBuilder()->create<VarExpr>();
+        varExpr->type = (QualType)getASTBuilder()->getTypeType(toType);
+        varExpr->declRef = isDeclRefTypeOf<Decl>(toType);
+
+        auto* constructorExpr = getASTBuilder()->create<InvokeExpr>();
+        constructorExpr->functionExpr = varExpr;
+        constructorExpr->arguments.addRange(coercedArgs);
+        constructorExpr->loc = loc;
+        auto resolvedConstructorExpr = CheckExpr(constructorExpr);
+
+        return resolvedConstructorExpr;
+    }
+
+    // translation from initializer list to constructor invocation if the struct has constructor.
+    bool SemanticsVisitor::_invokeExprForExplicitCtor(Type* toType, InitializerListExpr* fromInitializerListExpr, Expr** outExpr)
+    {
+        if(auto toDeclRefType = as<DeclRefType>(toType))
+        {
+            auto toTypeDeclRef = toDeclRefType->getDeclRef();
+            if(auto toStructDeclRef = toTypeDeclRef.as<StructDecl>())
```

Comment:
We need to unify the style, but most of the code we have puts a space between the keyword `if`, `for` etc. and the `(`.

Reply:
update this for every location I have touched.